### ISSUE TITLE
a question about code: how to deal with the unexpectedly failure of supervisor, as here is no **max_restarts**

### ISSUE THREAD

**Issue by hefeiyun** [ray-project/ray#41026]
### What happened + What you expected to happen

how to deal with the unexpectedly failure of supervisor, as here is no **max_restarts**

https://github.com/ray-project/ray/blob/master/dashboard/modules/job/job_manager.py
`supervisor = self._supervisor_actor_cls.options(
                lifetime="detached",
                name=JOB_ACTOR_NAME_TEMPLATE.format(job_id=submission_id),
                num_cpus=entrypoint_num_cpus,
                num_gpus=entrypoint_num_gpus,
                memory=entrypoint_memory,
                resources=entrypoint_resources,
                scheduling_strategy=scheduling_strategy,
                runtime_env=self._get_supervisor_runtime_env(
                    runtime_env, resources_specified
                ),
                namespace=SUPERVISOR_ACTOR_RAY_NAMESPACE,
            ).remote(submission_id, entrypoint, metadata or {}, self._gcs_address)`

### Versions / Dependencies

master

### Reproduction script

https://github.com/ray-project/ray/blob/master/dashboard/modules/job/job_manager.py

### Issue Severity

Low: It annoys or frustrates me.

**Comment by architkulkarni** [comment]
You're right that in the current design, if the supervisor actor fails, the job is failed.  Enabling `max_restarts` would require some thought around whether the constructor is idempotent and things like that.  

Unexpected actor failure should be rare though, how is it failing for you?

**Comment by hefeiyun** [comment]
> You're right that in the current design, if the supervisor actor fails, the job is failed. Enabling `max_restarts` would require some thought around whether the constructor is idempotent and things like that.
> 
> Unexpected actor failure should be rare though, how is it failing for you?

like this https://github.com/ray-project/ray/issues/41241 
would fail when no system resource
